integration_test/cases/type.exs

```diff
@@ -381,6 +403,7 @@ defmodule Ecto.Integration.TypeTest do
     assert [^decimal] = TestRepo.all(from p in Post, select: coalesce(p.cost, 0))
   end
 
+  @tag :union_with_literals
```

Comment:
Btw, why does this test fail? Is it because of number precision? :)

Reply:
that could be some odd bug. line 420 fails since somehow %{n: 1.0} is decoded into decimal. 

```
  1) test unions with literals (Ecto.Integration.TypeTest)
     c:/prj/elixir-ecto/ecto/integration_test/cases/type.exs:407
     ** (ArgumentError) cannot load `#Decimal<1.0>` as type :float
     code: assert TestRepo.all(union_all(query1, ^query2)) ==
     stacktrace:
       (ecto) lib/ecto/repo/queryable.ex:382: Ecto.Repo.Queryable.process/4
       (ecto) lib/ecto/repo/queryable.ex:367: anonymous fn/4 in Ecto.Repo.Queryable.process_kv/4
       (elixir) lib/enum.ex:1440: Enum."-map_reduce/3-lists^mapfoldl/2-0-"/3
       (ecto) lib/ecto/repo/queryable.ex:326: Ecto.Repo.Queryable.process/4
       (ecto) lib/ecto/repo/queryable.ex:246: anonymous fn/3 in Ecto.Repo.Queryable.postprocessor/4
       (elixir) lib/enum.ex:1336: Enum."-map/2-lists^map/1-0-"/2
       (ecto) lib/ecto/repo/queryable.ex:198: Ecto.Repo.Queryable.execute/4
       (ecto) lib/ecto/repo/queryable.ex:17: Ecto.Repo.Queryable.all/3
       c:/prj/elixir-ecto/ecto/integration_test/cases/type.exs:420: (test)

```